path_static_roles.go

```diff
@@ -213,18 +217,31 @@ func (b *backend) pathStaticRoleCreateUpdate(ctx context.Context, req *logical.R
 		}
 	}
 
-	dn := data.Get("dn").(string)
-	if dn == "" {
-		return logical.ErrorResponse("dn is a required field to manage a static account"), nil
-	}
-	role.StaticAccount.DN = dn
+	isCreate := req.Operation == logical.CreateOperation
+
+	b.managedUserLock.Lock()
+	defer b.managedUserLock.Unlock()
 
 	username := data.Get("username").(string)
-	if username == "" {
+	if isCreate && username == "" {
 		return logical.ErrorResponse("username is a required field to manage a static account"), nil
 	}
+	if isCreate && b.isManagedUser(username) {
+		return logical.ErrorResponse("%q is already managed by the secrets engine", username), nil
+	}
+	if !isCreate && username != role.StaticAccount.Username {
+		return logical.ErrorResponse("cannot update static account username"), nil
+	}
```

Comment:
Seems like there is a bit of test coverage missing for these conditions?

Reply:
Good catch -- will add coverage for this!

Reply:
Added coverage for this in https://github.com/hashicorp/vault-plugin-secrets-openldap/pull/40/commits/60d24c919929149777d6f6c264c7682a8c7dbc65. Thanks!